src/commands/test/runner.py

```diff
@@ -47,6 +52,11 @@
         match_pattern: Optional[Pattern] = None,
         omit_pattern: Optional[Pattern] = None,
     ):
+        original_run_from_entrypoint = CairoFunctionRunner.run_from_entrypoint
```

Comment:
can we make it more generic, i.e. as an instance's context manager?

Reply:
Initially, I tried to use subclass approach but `@replace_class` decorator didn't work, and I have to somehow pass an instance of `TestRunner` to `CairoFunctionRunner`.

Reply:
Yess It will be a bit tricky, but we won't get away with this solution long term. It won't work for isolated states and won't parallelize. As I said imo we can merge it for now, I will let you know if I manage to figure out something better ;)

Reply:
Ok, I am leaving this code for the time being.